### DISCUSSION TITLE
Pagination and automatic page generation question.

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#66565] [Help] [answered]
### Summary

For example, I have this page here:
https://github.com/Elindo586/nextprojectb/blob/main/pages/video-blog/english/index.js

^^ This is a paginated page that shows certain amount of items, however, it doesn't have a web address for each paginated set of items, so when I click the back button on the browser, it doesn't take me back to the previous set of items, instead, it takes me out of the entire page.

Having a web address per each page would be convenient to send a link for a a group of items, also would help on not losing track of an specific group of items if somebody hits the back button on the browser.

I was wondering if Next.js has some tools that can be used for these type of operations?

**Comment by icyJoseph** [comment, marked as answer]
Put the current page in the URL as a query, and use that as your source of truth, rather than react state. Make sure you parse the page name from the URL, and clamp it to a limit that makes sense for your use case. Like if someone were to request page 100000 you should know that, you should make sure your UI doesn't get tricked by that.

If there's no page in the url query, then you start at the first page.

You can also maybe do `/pages/video-blog/english/[[...page]].js` which uses optional catch all, so that the URL looks different, but it is almost the same as the idea up above.

- https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes#optional-catch-all-segments

If the page value is empty list, start at the beginning.

**Comment by Elindo586** [comment]
Cool beans... 

I will check it out..
